### PR TITLE
`Development`: Add JavaDocs to UtilServices and Factories [9]

### PR DESCRIPTION
**Pull request by DominikRemo**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

Improve the readability of UtilServices and Factories by adding JavaDoc to the helper methods

### Description

Added documentation to following files:
- [x] exercise/textexercise/TextExerciseFactory.java
- [x] exercise/textexercise/TextExerciseUtilService.java

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/exercise/textexercise/TextExerciseUtilService.java [hunk 1]
```diff
@@ -81,10 +81,10 @@ public class TextExerciseUtilService {
...
     /**
-     * Generate a set of specified size containing TextBlocks with dummy Text
+     * Creates a Set with a given number of TextBlocks.
      *
-     * @param count expected size of TextBlock set
-     * @return Set of dummy TextBlocks
+     * @param count The number of TextBlocks to create
+     * @return The new Set of dummy TextBlocks
      */
     public Set<TextBlock> generateTextBlocks(int count) {
```
Comment (laadvo) [inline src/test/java/de/tum/in/www1/artemis/exercise/textexercise/TextExerciseUtilService.java:89]: Could you move this method into `TextExerciseFactory`?

File: src/test/java/de/tum/in/www1/artemis/exercise/textexercise/TextExerciseUtilService.java [hunk 2]
```diff
@@ -318,11 +438,26 @@ public TextPlagiarismResult createTextPlagiarismResultForExercise(Exercise exerc
         return plagiarismResultRepo.save(result);
     }
 
+    /**
+     * Creates a TextAssessmentEvent for the given parameters.
```
Comment (laadvo) [inline src/test/java/de/tum/in/www1/artemis/exercise/textexercise/TextExerciseUtilService.java:442]:
```suggestion
     * Creates a TextAssessmentEvent with the given parameters.
```
I think with sounds better here.

File: src/test/java/de/tum/in/www1/artemis/exercise/textexercise/TextExerciseUtilService.java [hunk 3]
```diff
@@ -346,12 +481,13 @@ public TextExercise createTextExerciseForExam(ExerciseGroup exerciseGroup) {
     }
 
     /**
-     * Creates a submission with a result and an assessor of the passed text exercise.
+     * Creates and saves a TextSubmission and StudentParticipation for the given TextExercise, and studentLogin. Also creates and saves a Result for the
```
Comment (laadvo) [inline src/test/java/de/tum/in/www1/artemis/exercise/textexercise/TextExerciseUtilService.java:484]: Could you mention the tutor Assessor here?

**Review by laadvo — COMMENTED** [review]
Already looks good, left some minor remarks

File: src/test/java/de/tum/in/www1/artemis/exercise/textexercise/TextExerciseUtilService.java (hunk 3 above)
Comment (laadvo) [reply src/test/java/de/tum/in/www1/artemis/exercise/textexercise/TextExerciseUtilService.java:467]: could you also update it in the beginning? 
```suggestion
     * Creates and saves a TextSubmission and StudentParticipation for the given TextExercise, and studentLogin. Also creates and saves a Result for the StudentParticipation given the assessorLogin.
```

Comment (DominikRemo) [reply src/test/java/de/tum/in/www1/artemis/exercise/textexercise/TextExerciseUtilService.java:467]: Done, thanks for noticing :)

**Review by laadvo — APPROVED** [review]
New JavaDoc LGTM :)

**Review by b-fein — APPROVED** [review]

**Pull request merged**